libbeat/dashboards/export.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dashboards
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/elastic/beats/filebeat/scripts/generator"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/kibana"
+)
+
+// ListYML is the yaml file which contains list of available dashboards.
+type ListYML struct {
+	Dashboards []YMLElement `yaml:"dashboards"`
+}
+
+// YMLElement contains the data of a dashboard:
+// * its uuid in Kibana
+// * filename to be saved as
+type YMLElement struct {
+	ID   string `yaml:"id"`
+	File string `yaml:"file"`
+}
+
+// Export wraps GetDashboard call to provide a more descriptive API
+func Export(client *kibana.Client, id string) (common.MapStr, error) {
+	return client.GetDashboard(id)
+}
+
+// ExportAllFromYml exports all dashboards found in the YML file
+func ExportAllFromYml(client *kibana.Client, yml string) ([]common.MapStr, ListYML, error) {
```

Comment:
suggestion to rename yml to ymlFile or ymlPath.
suggestion to split ExportAllFromYML into two functions.

ExportAllFromYaml(client *kibana.Client, ymlFile) -> Read the yaml file and unmarshal it.
ExportAll(client *kibana.Client, list *ListYaml) -> receive a listYaml.


This would give us more flexibility, and no need to create a yml file for most of the testing.